src/file_binding.ml

```diff
@@ -37,6 +37,15 @@ module Unexpanded = struct
 
   let expand_src t ~dir ~f = Path.relative dir (f t.src)
 
+  let destination_relative_to_install_path t ~f =
+    let dst = Option.map ~f t.dst in
+    match Install.Entry.adjust_dst_on_windows
+            ~src_basename:(fun () -> Filename.basename (f t.src))
+            ~dst
```

Comment:
Discussed with @aalekseyev in person: currently this imposes more restrictions on what variables can be used on Windows only, which is not great